core/server/middleware/check-ssl.js

```diff
@@ -44,7 +44,7 @@ checkSSL = function checkSSL(req, res, next) {
                 forceAdminSSL: config.get('forceAdminSSL'),
                 configUrlSSL: config.get('urlSSL'),
                 configUrl: config.get('url'),
-                reqUrl: req.url
+                reqUrl: req.originalUrl || req.url
```

Comment:
Just added another TODO to check the filterPrivateRoutes middleware in the private-blogging app because this is also using req.url and I'm not sure if the usage is correct.